src/modules/Game.class.js

```diff
@@ -50,19 +91,121 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.status = 'playing';
+    this.addRandomTile();
+    this.addRandomTile();
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.status = 'idle';
+    this.board = this.boardDeepCopy(this.initialState);
+    this.score = 0;
+  }
 
   // Add your own methods here
+
+  addRandomTile() {
+    const emptyCells = this.board.reduce((acc, row, rowIndex) => {
+      row.forEach((cell, colIndex) => {
+        if (cell === 0) {
+          acc.push({ row: rowIndex, col: colIndex });
+        }
+      });
+
+      return acc;
+    }, []);
+
+    if (emptyCells.length > 0) {
+      const { row, col } =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      this.board[row][col] = Math.random() < 0.9 ? 2 : 4;
+    }
+
+    if (this.isGameOver()) {
+      this.status = 'lose';
+    }
+  }
+
+  joinTiles(board) {
+    return board.map((line) => {
+      const filteredLine = line.filter((value) => value !== 0);
+
+      for (let i = 0; i < filteredLine.length; i++) {
+        if (filteredLine[i] === filteredLine[i + 1]) {
+          filteredLine[i] *= 2;
+          this.score += filteredLine[i];
+
+          if (filteredLine[i] === 2048) {
+            this.status = 'win';
+          }
+
+          filteredLine.splice(i + 1, 1);
+        }
+      }
+
+      while (filteredLine.length < 4) {
+        filteredLine.push(0);
+      }
+
+      return filteredLine;
+    });
+  }
+
+  transpose(board) {
+    return board[0].map((_, colInd) => board.map((row) => row[colInd]));
+  }
+
+  reverse(board) {
+    return board.map((line) => line.reverse());
+  }
+
+  boardDeepCopy(toCopy) {
+    return JSON.parse(JSON.stringify(toCopy));
+  }
+
+  isGameOver() {
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        const current = this.board[i][j];
+
+        if (
+          this.board[i][j] === 0 ||
+          (i < 3 && current === this.board[i + 1][j]) ||
+          (j < 3 && current === this.board[i][j + 1])
+        ) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  }
+
+  performMove(moveFunction) {
+    if (this.status !== 'playing') {
+      return;
+    }
```

Comment:
The `performMove` method checks if the game status is 'playing' before proceeding. This is good for preventing moves when the game is not active. Ensure that the game status is correctly updated to 'win' or 'lose' when those conditions are met, so that this check functions correctly.